### PR TITLE
ARP: implement decoder and logger

### PR DESCRIPTION
**Pull request by glongo**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at
   https://docs.suricata.io/en/latest/devguide/contributing/contribution-process.html
- [x] I have signed the Open Information Security Foundation contribution agreement at
   https://suricata.io/about/contribution-agreement/ (note: this is only required once)
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/6827

Describe changes:
- Add decoder and logger for ARP protocol
- Make `JSONFormatAndAddMACAddr` public
- Update doc

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1681

### CODE REVIEW

File: src/output-json-arp.c [hunk 1]
```diff
@@ -0,0 +1,256 @@
+/* Copyright (C) 2021 Open Information Security Foundation
```
Comment (jlucovsky) [inline src/output-json-arp.c:1]: nit: year

File: src/output-json-arp.c [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+            return "infiniband";
+        case 33:
+            return "cai";
+        case 34:
+            return "wiegand_interface";
+        case 35:
+            return "pure_ip";
+        case 36:
+            return "hw_exp1";
+        case 37:
```
Comment (jlucovsky) [inline src/output-json-arp.c:131]: https://www.iana.org/assignments/arp-parameters/arp-parameters.xhtml

38 has a value as do 256 and 257.

File: src/output-json-arp.c [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    jb_close(jb); /* arp */
+    OutputJsonBuilderBuffer(jb, thread);
+    jb_free(jb);
+
+    return TM_ECODE_OK;
+}
+
+static bool JsonArpLogCondition(ThreadVars *tv, void *thread_data, const Packet *p)
+{
+    if (!PKT_IS_ARP(p)) {
```
Comment (jlucovsky) [inline src/output-json-arp.c:242]: nit: suggest `return PKT_IS_ARP(p)`

File: src/output-json-arp.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
+/* Copyright (C) 2021 Open Information Security Foundation
```
Comment (jlucovsky) [inline src/output-json-arp.h:1]: nit: year

**Comment by glongo** [comment]
Replaced with https://github.com/OISF/suricata/pull/10572

**Pull request closed without merging**